### PR TITLE
fix(bn-checkbox): pass disabled prop to input

### PR DESCRIPTION
**Pull request by gmq**
## Context
Banano is a component library intended for use in Platanus projects. The BnCheckbox component wraps a `type="checkbox"` input, allowing the use of v-model and vee-validate. 

## This PR

This PR fixes the missing disabled attribute in the input element. It also adds a default value of `true` to the checkbox and also adds supports for the `uncheckedValue` vee-validate feature.

In addition, the PR updates some libraries and adds documentation for BnCheckbox.

### CODE REVIEW

**Comment by gmq** [comment]
[Cuando el checkbox está disabled, se puede clickear y hacer focus igual](https://trello.com/c/JPEbdxVV/25-cuando-el-checkbox-est%C3%A1-disabled-se-puede-clickear-y-hacer-focus-igual)

File: src/components/BnCheckbox/BnCheckbox.vue [hunk 1]
```diff
@@ -9,7 +9,8 @@ interface Props {
   value: valueTypes,
   name: string,
   color?: string,
-  rules?: RuleExpression<unknown>,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  rules?: RuleExpression<any>,
```
Comment (fprebolledo) [inline src/components/BnCheckbox/BnCheckbox.vue]: Here we could change this to what we discussed in PR [11](https://github.com/platanus/banano/pull/11)

File: src/components/BnCheckbox/BnCheckbox.vue [hunk 2]
```diff
@@ -30,6 +32,7 @@ const { handleChange, checked, meta, setTouched } = useField(props.name, props.r
   type: 'checkbox',
   checkedValue: props.value,
   initialValue: props.modelValue,
+  uncheckedValue: props.uncheckedValue,
```
Comment (fprebolledo) [inline src/components/BnCheckbox/BnCheckbox.vue]: Wouldn't this have to be a reference?

**Review by fprebolledo — APPROVED** [review]
It looks OK. I left a comment related to the `RulesExpression`.

File: src/components/BnCheckbox/BnCheckbox.vue (hunk 2 above)
Comment (gmq) [reply src/components/BnCheckbox/BnCheckbox.vue]: True. I just added a commit fixing this. Checking this https://github.com/logaretm/vee-validate/blob/main/packages/vee-validate/src/Field.ts#L119C1-L123C49 it seems that almost everything that is used in `useField` should be transformed into a ref first.

File: src/components/BnCheckbox/BnCheckbox.vue [hunk 3]
```diff
@@ -1,34 +1,38 @@
...
 
 interface Props {
-  modelValue?: valueTypes | valueTypes[],
-  value: valueTypes,
+  modelValue?: ValueType | ValueType[],
+  value?: ValueType,
   name: string,
   color?: string,
-  rules?: RuleExpression<unknown>,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (fprebolledo) [inline src/components/BnCheckbox/BnCheckbox.vue]: It is still necessary to disable this rule?

**Review by fprebolledo — APPROVED** [review]
:gem:

**Pull request merged**